stdlib/public/core/StringBridge.swift

```diff
@@ -234,35 +243,143 @@ internal func _bridgeCocoaString(_ cocoaString: _CocoaString) -> _StringGuts {
     //   2) If it's mutable with no associated information, then a copy must
     //      happen; might as well eagerly bridge it in.
     //   3) If it's mutable with associated information, must make the call
-    let immutableCopy
-      = _stdlib_binary_CFStringCreateCopy(cocoaString) as AnyObject
+    let immutableCopy =
+      _stdlib_binary_CFStringCreateCopy(cocoaString) as AnyObject
     
-#if !(arch(i386) || arch(arm))
+    #if !(arch(i386) || arch(arm))
     if _isObjCTaggedPointer(immutableCopy) {
       return _StringGuts(_SmallString(taggedCocoa: immutableCopy))
     }
-#endif
+    #endif
     
     let (fastUTF8, isASCII): (Bool, Bool)
     switch _getCocoaStringPointer(immutableCopy) {
     case .ascii(_): (fastUTF8, isASCII) = (true, true)
     case .utf8(_): (fastUTF8, isASCII) = (true, false)
     default:  (fastUTF8, isASCII) = (false, false)
     }
-    let length = _stdlib_binary_CFStringGetLength(immutableCopy)
     
     return _StringGuts(
       cocoa: immutableCopy,
       providesFastUTF8: fastUTF8,
       isASCII: isASCII,
       length: length)
+}
+
+@_effects(releasenone)
+public //SPI(Foundation)
+func _bridgeConstantCocoaString(
+  _ cocoaString: __owned AnyObject, _ length: Int) -> String {
+  let (fastUTF8, isASCII): (Bool, Bool)
+  switch _getCocoaStringPointer(cocoaString) {
+  case .ascii(_): (fastUTF8, isASCII) = (true, true)
+  case .utf8(_): (fastUTF8, isASCII) = (true, false)
+  default:  (fastUTF8, isASCII) = (false, false)
   }
+  
+  return String(_StringGuts(
+    cocoa: cocoaString,
+    providesFastUTF8: fastUTF8,
+    isASCII: isASCII,
+    length: length))
+}
+
+@_effects(releasenone)
+public //SPI(Foundation)
+func _bridgeUnknownCocoaString(
+  _ cocoaString: __owned AnyObject, _ length: Int) -> String {
+ return _bridgeConstantCocoaString(cocoaString, length)
+}
+
+@_effects(readonly)
+public //SPI(Foundation)
+func _isRebridgedSwiftStringClass(_ cls: AnyClass) -> Bool {
+  return cls == __StringStorage.self || cls == __SharedStringStorage.self
+}
+
+@_effects(releasenone)
+public //SPI(Foundation)
+func _rebridgeSwiftString(
+  _ cocoaString: __owned AnyObject, _ cls: AnyClass) -> String {
+  _internalInvariant(_isRebridgedSwiftStringClass(cls))
+  if cls == __StringStorage.self {
+    return String(_unsafeUncheckedDowncast(
+      cocoaString, to: __StringStorage.self).asString._guts)
+  }
+  return String(_unsafeUncheckedDowncast(
+    cocoaString, to: __SharedStringStorage.self).asString._guts)
+}
+
+@_effects(releasenone)
+public //SPI(Foundation)
+func _bridgeTaggedCocoaString(_ cocoaString: AnyObject) -> String? {
+  #if !(arch(i386) || arch(arm))
+  if _isObjCTaggedPointer(cocoaString) {
```

Comment:
This check would be faster in the caller but we can't use builtins or _isObjCTaggedPointer in the Foundation overlay :(